### PR TITLE
feat: add race round implementation

### PR DESCRIPTION
**Pull request by Tran-Antoine**
Adds a first version of an implementation for the so called "race round".
A race round is a `GameRound` that is initialized with one question, and first to answer correctly gets all the points, other players get 0. 

Properties that may be configurated through the builder of the object:
- points to be awarded to the single winner
- max guesses allowed per player

Ending conditions:
- one player answered 100% correctly
or
- all players ran out of guesses

Game report:
- winner ID
- points awarded

### CODE REVIEW

File: api/src/main/java/net/starype/quiz/api/game/AnswerEligibility.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return players.stream().anyMatch(this::isEligible);
+    }
+
+    default AnswerEligibility or(AnswerEligibility other) {
+        return (player) -> this.isEligible(player) || other.isEligible(player);
+    }
+
+    default AnswerEligibility and(AnswerEligibility other) {
+        return (player) -> this.isEligible(player) && other.isEligible(player);
+    }
```
Comment (mredstone6) [inline api/src/main/java/net/starype/quiz/api/game/AnswerEligibility.java:20]: The name of this class suggest that it perform a check if the answer is correctly formatted. So it doesn't make sense that it takes UUIDHolder as input. You could rename this ```EntityEligibility```

File: api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        guessesPerPlayer.put(holder, guessesPerPlayer.getOrDefault(holder, 0) + 1);
+    }
+
+    public void giveUp(UUIDHolder holder) {
+        guessesPerPlayer.put(holder, maxGuesses);
+    }
+
+    @Override
+    public boolean isEligible(UUIDHolder player) {
+        return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses;
```
Comment (mredstone6) [inline api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java:28]: Adding a condition if the ```maxGuesses``` is Integer.MAX_VALUE then always return true (use or)

**Review by mredstone6 — CHANGES_REQUESTED** [review]

File: api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java (hunk 2 above)
Comment (Tran-Antoine) [reply api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java:28]: Yes I thought about maybe using `-1` as the value for unlimited tries

Comment (mredstone6) [reply api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java:28]: I think using the MAX_VALUE is a better approaches as it is really a big value but your idea isn't bad. If so then add an
```java
return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses || maxGuesses == -1;
return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses || maxGuesses == Integer.MAX_VALUE;
```

**Review by mredstone6 — APPROVED** [review]

**Review by mredstone6 — CHANGES_REQUESTED** [review]
Again the issue with the MaxGuessCounter (for an Infinite amount of Guesses)

**Review by mredstone6 — APPROVED** [review]

**Pull request merged**